overlord/configstate/configstate.go

```diff
@@ -47,21 +47,46 @@ func ConfigureHookTimeout() time.Duration {
 	return timeout
 }
 
+func canConfigure(st *state.State, snapName string) error {
+	// the "core" snap/pseudonym can always be configured as it
+	// is handled internally
+	if snapName == "core" {
+		return nil
+	}
+
+	var snapst snapstate.SnapState
+	err := snapstate.Get(st, snapName, &snapst)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+
+	if !snapst.IsInstalled() {
+		return &snap.NotInstalledError{Snap: snapName}
+	}
+
+	// the "snapd" snap cannot be configured yet
+	if snapName == "snapd" {
+		return fmt.Errorf(`cannot configure the "snapd" snap, please use "system" instead`)
```

Comment:
I think we need to discuss but yes, that is definitely an option. We need to discuss a bit more what to do, there will be a followup with config anyway because right now snapd assumes the snap (core) that has config is installed. We need to remove this restriction. So we might hang all config into "system" or "core". I think its just a matter of discussing, for now this is the most conservative option :)